src/utils/hooks/useCapturedEvents.ts

```diff
@@ -1,17 +1,20 @@
 /* eslint-disable react-hooks/exhaustive-deps */
 import { useEffect, useCallback } from 'react'
+import { or } from '../../components/custom-ui-helper'
+import React from 'react'
 
 /**
  * ! Call this hook inside Shadow Root!
  */
 export function useCapturedInput(
-    ref: React.MutableRefObject<HTMLInputElement | undefined | null>,
     onChange: (newVal: string) => void,
     deps: any[] = [],
+    _ref?: React.MutableRefObject<HTMLInputElement | undefined | null>,
 ) {
+    const ref = or(_ref, React.useRef(null))
     const stop = useCallback((e: Event) => e.stopPropagation(), deps)
     const use = useCallback(
-        (e: Event) => onChange((e.currentTarget as HTMLInputElement).value),
+        (e: Event) => onChange((e.currentTarget as HTMLInputElement)?.value ?? (e.target as HTMLInputElement)?.value),
```

Comment:
Why this necessary here?

Reply:
`currentTarget` and `target` some works for `Input`(mui) some works for the native input